feat: add `blas/ext/base/gfind-index` (#7727)
stdlib-js/stdlib: lib/node_modules/@stdlib/blas/ext/base/gfind-index/README.md

<!--

@license Apache-2.0

Copyright (c) 2025 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# gfindIndex

> Return the index of the first element which passes a test implemented by a predicate function.

<section class="usage">

## Usage

```javascript
var gfindIndex = require( '@stdlib/blas/ext/base/gfind-index' );
```

#### gfindIndex( N, x, strideX, clbk\[, thisArg] )

> Returns the index of the first element which passes a test implemented by a predicate function.

Author: ```suggestion
Returns the first index of an element which passes a test implemented by a predicate function.
```
@kgryte Is this suggestion okay or the original description?

Reviewer: The original description is fine.

```javascript
function isEven( v ) {
    return v % 2.0 === 0.0;
}

var x = [ 1.0, 3.0, -5.0, 4.0, 0.0, -1.0, -3.0 ];

var idx = gfindIndex( x.length, x, 1, isEven );
// returns 3
```

> If the function is unable to find an element, the function returns `-1`.

Reviewer: ```suggestion
If no element passes a test implemented by a predicate function, the function returns `-1`.
```

```javascript
function isEven( v ) {
    return v % 2.0 === 0.0;
}

var x = [ 1.0, 3.0, 5.0, 7.0, 9.0 ];

var idx = gfindIndex( x.length, x, 1, isEven );
// returns -1
```

The function has the following parameters:

-   **N**: number of indexed elements.
-   **x**: input array.
-   **strideX**: stride length.

> -   **predicate**: predicate function.

Author: ```suggestion
-   **clbk**: callback function.
```

-   **thisArg**: execution context (_optional_).

The callback function is provided the following arguments:

-   **value**: current array element.
-   **aidx**: array index.
-   **sidx**: strided index (`offset + aidx*stride`).
-   **array**: the input array.

To set the callback execution context, provide a `thisArg`.

```javascript
function isEven( v ) {
    this.count += 1;
    return v % 2.0 === 0.0;
}

var x = [ 1.0, 3.0, -5.0, 4.0, 0.0, -1.0, -3.0 ];

var ctx = {
    'count': 0
};

var idx = gfindIndex( x.length, x, 1, isEven, ctx );
// returns 3

var count = ctx.count;
// returns 4
```

> The `N` and stride parameters determine which elements in the strided array are accessed at runtime. For example, to check every other element:

Reviewer: ```suggestion
The `N` and stride parameters determine which elements in the strided array are accessed at runtime. For example, to test every other element:
```

```javascript
function isEven( v ) {
    return v % 2.0 === 0.0;
}

var x = [ 1.0, 3.0, -5.0, 0.0, 4.0, -1.0, -3.0 ];

var idx = gfindIndex( 4, x, 2, isEven );
// returns 2
```

Note that indexing is relative to the first index. To introduce an offset, use [`typed array`][mdn-typed-array] views.

```javascript
var Float64Array = require( '@stdlib/array/float64' );

function isEven( v ) {
    return v % 2.0 === 0.0;
}

// Initial array...
var x0 = new Float64Array( [ 1.0, 3.0, 4.0, -5.0, 7.0, 6.0 ] );

> // Create an offset view...

Reviewer: ```suggestion
// Initial array:
var x0 = new Float64Array( [ 1.0, 3.0, 4.0, -5.0, 7.0, 6.0 ] );

// Create an offset view:
```
